src/mmrelay/meshtastic_utils.py

```
@@ -1003,6 +1102,8 @@ def call_get_metadata() -> None:
                 # A previous metadata request is still running; avoid piling up
                 # threads and leave the in-flight call to finish in its own time.
                 logger.debug("getMetadata() already running; skipping new request")
+                if raise_on_error:
+                    raise RuntimeError("getMetadata() probe already in progress")
```

Ruff (0.15.4)

[warning] Abstract `raise` to an inner function

(TRY301)